File: tests/fixtures/generic/clean/taint_xxe_clean.cs
// @audit-fixture
// @rule: taint_xxe
// @category: security
// @expected: clean
using System.Xml;
using System.Xml.Linq;
public class XmlController : Controller {
    public IActionResult Parse() {
        var settings = new XmlReaderSettings { DtdProcessing = DtdProcessing.Prohibit, XmlResolver = null };
        var doc = XDocument.Load(XmlReader.Create(new StringReader(body), settings));
        return Ok();
    }
}

File: tests/fixtures/generic/clean/taint_xxe_clean.js
// @audit-fixture
// @rule: taint_xxe
// @category: security
// @expected: clean
const { parseXml } = require('libxmljs2');
const express = require('express');
const app = express();
app.post('/parse', (req, res) => {
    const doc = parseXml(req.body.xml, { noent: false, nonet: true });
    res.json({ ok: true });
});
